website/app/components/doc/page/header/algolia-search/index.js

```diff
@@ -240,11 +243,10 @@ export default class DocAlgoliaSearchComponent extends Component {
       });
     }
 
-    // add event listener to open/close the modal via `command-K` hot-key
     document.addEventListener('keydown', (event) => {
+      // add event listener to open/close the modal via `command-K` hot-key
       if (event.metaKey && event.key === 'k') {
         autocompleteInstance.setIsOpen(!this.isModalOpen);
-        this.isModalOpen = !this.isModalOpen;
```

Comment:
Note: I moved this to the onStateChange argument on line 60 because if you used cmd+k after opening the dialog with the search button or closing it with escape, the tracked state wasn't properly updated. Now- no matter how the user opens/closes the dialog the tracked state is always correct.